internal/restorable/image.go

```diff
@@ -720,34 +713,29 @@ func regionToRectangle(region graphicsdriver.Region) image.Rectangle {
 		int(math.Ceil(float64(region.Y+region.Height))))
 }
 
-// removeDuplicatedRegions removes duplicated regions and returns the new size of the slice.
-// If a region covers other regions, the covered regions are removed.
-func removeDuplicatedRegions(regions []image.Rectangle) int {
-	for i, r := range regions {
-		if r.Empty() {
-			continue
-		}
-		for j, rr := range regions {
-			if i == j {
-				continue
-			}
-			if rr.Empty() {
-				continue
-			}
-			if rr.In(r) {
-				regions[j] = image.Rectangle{}
-			}
+// appendRegionRemovingDuplicates adds a region to a given list of regions,
+// but removes any duplicate between the newly added region and any existing regions.
+//
+// In case the newly added region is fully contained in any pre-existing region, this function does nothing.
+// Otherwise, any pre-existing regions that are fully contained in the newly added region are removed.
+//
+// This is done to avoid unnecessary reading pixels from GPU.
+func appendRegionRemovingDuplicates(regions []image.Rectangle, region image.Rectangle) []image.Rectangle {
```

Comment:
Mutating the argument `regions`'s content and returning the same slice seems odd. Let's return the new length of the slice as the original `removeDuplicatedRegions` does.

Reply:
Replaced by pointer now as requested.